### ISSUE TITLE
Hammerspoon hanging for a while (sometimes) when invoking Seal

### ISSUE THREAD

**Issue by zzamboni** [Hammerspoon/hammerspoon#2243]
Sometimes, when I invoke Seal with its keybinding, but also often when clicking on the HS menubar item, Hammerspoon seems to hang for a while. It usually comes back, but sometimes after 1-2 minutes of beachball.

I captured a sample using Activity Monitor the last time this happened: https://gist.github.com/zzamboni/c47e291aa4d41d5db532f1246969620e

I'm not an expert in interpreting these, but AFAICT there seems to be near the top, something calling a synchronous HTTP request - maybe this is causing the blocking? I'll look through my config and try to report back.

In the meantime, I'd appreciate any further insights from this.

**Comment by zzamboni** [comment]
I think I have located the culprit: the [Leanpub](http://www.hammerspoon.org/Spoons/Leanpub.html) spoon, which I am using and which uses synchronous `hs.http.get()` calls, executed regularly from a timer. I have disabled it in my config and the slowdowns seem to be gone. I'll update the spoon to use asynchronous calls and try again.

I'm leaving this issue open for now, until I confirm and fix the culprit.
